src/bin/pg_autoctl/cli_common.c

```diff
@@ -401,6 +513,79 @@ cli_create_node_getopts(int argc, char **argv,
 }
 
 
+/*
+ * cli_getopt_ssl_flags parses the SSL related options from the command line.
+ *
+ * { "ssl-ca-file", required_argument, &ssl_flag, SSL_CA_FILE_FLAG }
+ * { "ssl-crl-file", required_argument, &ssl_flag, SSL_CRL_FILE_FLAG }
+ * { "server-crt", required_argument, &ssl_flag, SSL_SERVER_CRT_FLAG }
+ * { "server-key", required_argument, &ssl_flag, SSL_SERVER_KEY_FLAG }
+ * { "ssl-mode", required_argument, &ssl_flag, SSL_MODE_FLAG },
+ *
+ * As those options are not using any short option (one-char) variant, they all
+ * fall in the case 0, and we can process them thanks to the global variable
+ * ssl_flag, an int.
+ */
+bool
+cli_getopt_ssl_flags(PostgresSetup *pgSetup)
```

Comment:
can we have ssl_flag passed as argument instead of being used as global variable ?

Reply:
and optarg too

Reply:
I could do some of it as suggested, but my C compiler wants to know the size of the ssl_flag pointer at compile time and that means global variable as far as I can see.